agent/consul/acl_endpoint.go

```diff
@@ -1324,11 +1324,15 @@ func (a *ACL) GetPolicy(args *structs.ACLPolicyResolveLegacyRequest, reply *stru
 	// Get the policy via the cache
 	parent := a.srv.config.ACLDefaultPolicy
 
-	policy, err := a.srv.acls.GetMergedPolicyForToken(args.ACL)
+	ident, policy, err := a.srv.acls.GetMergedPolicyForToken(args.ACL)
 	if err != nil {
 		return err
 	}
 
+	if token, ok := ident.(*structs.ACLToken); ok && token.Type == structs.ACLTokenTypeManagement {
+		parent = "manage"
```

Comment:
Is this the only special case we'd need to consider?

Reply:
Yes, client type tokens should use the default policy as the parent instead of overriding with "manage". I went back and looked at the v1.2 code and there was a specific check for a management token where it would override the "parent" policy similarly.

For normal client tokens the rules of the token will get encoded into the policy structure.